go/libraries/doltcore/schema/index_coll.go

```diff
@@ -64,6 +64,8 @@ type IndexCollection interface {
 	RenameIndex(oldName, newName string) (Index, error)
 	//SetPks changes the pks or pk ordinals
 	SetPks([]uint64) error
+	// SetIndexPrefixLength sets the prefix lengths for the specified index
+	SetIndexPrefixLength(indexName string, prefixLengths []uint16) error
```

Comment:
This is the wrong abstraction here. `SetPks` applies to the index collection, which is modifiable by nature.  This is not the case for indexes, especially having a function on the collection to modify indexes. This needs to be moved to index creation.